### PR TITLE
Server-side (reader) support for non-synchronizing literals

### PR DESCRIPTION
**Pull request by foxcpp**
Not sure if we want something for the client side.

### CODE REVIEW

File: read.go [hunk 1]
```diff
@@ -234,6 +234,12 @@ func (r *Reader) ReadLiteral() (Literal, error) {
 		return nil, err
 	}
 	lstr = trimSuffix(lstr, literalEnd)
+
+	nonSync := lstr[len(lstr)-1] == '+'
```
Comment (emersion) [inline read.go]: This doesn't check whether the string is non-empty. We could use `strings.HasSuffix` to prevent issues like this.

Comment (foxcpp) [reply read.go]: Oops.

**Comment by emersion** [comment]
>Not sure if we want something for the client side.

Client-side support is probably desirable. Can be left as a follow-up issue though.

**Comment by foxcpp** [comment]
Aw, don't write code while being distracted!

**Comment by emersion** [comment]
Thanks!

**Pull request merged**